### DISCUSSION TITLE
How to use Pixi InteractionManager setTargetElement

### DISCUSSION THREAD

**Discussion by joel1st** [pixijs/pixijs#7615] [Q&A]
Hello,
I'm running into an interesting situation where setTargetElement is working (in the sense that the debugger is showing that it is receiving events from the dom element on top), but the events are not triggering interactive behavior in the scene.

How the project is setup:
- I have a three.js and pixi.js project rendering on separate canvases but positioned on top of each other, the Three.js canvas is rendering directly above the pixi.js canvas (both are position: absolute, top: 0, left: 0, width: 500px, height: 500px). 

- If I remove the three.js canvas from the top (and remove the setTargetElement), the interactive events are handled properly (eg a sprite with interactive = true and a 'pointerdown' sprite event, triggers the event when I mouse down). 

- If I keep the three.js element on top and set it as the interactive element with 'setTargetElement' `app.renderer.plugins.interaction.setTargetElement(threeCanv,1)` it triggers this event: https://pixijs.download/release/docs/packages_interaction_src_InteractionManager.ts.html (onPointerDown function), but it no longer triggers the sprite event.

Is this intended behaviour or a bug?

PIXI v6.0.4

**Comment by ivanpopelyshev** [comment]
if you dont want threejs to do any events at all - you can use `pointer-events: none;` css on that canvas.

If your case is different, you hav to debug more or give us full reproduction. It could be that I already had an answer, but not in this case :(

**Comment by joel1st** [comment]
Hey Ivan, thanks for the reply.

Looks like I can't reproduce the issue in a simple case, so must be something on my side. For anyone else trying to achieve the same thing, here is the attempt at reproduction:

https://codesandbox.io/s/pixijs-forked-x615y?file=/src/index.js

**Comment by joel1st** [comment]
Actually I've managed to reproduce it by using a regular div instead of three.js:
https://codesandbox.io/s/pixijs-forked-lf3db?file=/src/index.js

**Comment by andrewstart** [comment]
I don't know why the three.js canvas wouldn't work, but it looks like the math in [mapPositionToPoint](https://github.com/pixijs/pixijs/blob/dev/packages/interaction/src/InteractionManager.ts#L1233-L1234) is expecting to be used on a canvas. I'm guessing that we should be using the Pixi Renderer's canvas instead of `(this.interactionDOMElement as any)`, but I'm not sure.

**Comment by joel1st** [comment]
> I don't know why the three.js canvas wouldn't work, but it looks like the math in [mapPositionToPoint](https://github.com/pixijs/pixijs/blob/dev/packages/interaction/src/InteractionManager.ts#L1233-L1234) is expecting to be used on a canvas. I'm guessing that we should be using the Pixi Renderer's canvas instead of `(this.interactionDOMElement as any)`, but I'm not sure.

Yeah I managed to get it working when applying it directly to the three.js canvas. I have not had any luck for non-canvas elements. This might be intended behaviour, in which case setTargetElement documentation/types should take a HTMLCanvasElement instead of a generic HTMLElement

**Comment by bigtimebuddy** [comment]
This is more of a support question than a bug. Supporting three.js and pixi on top of each other with interactivity is not a case we're supporting. Converting this to a discussion.
